### PR TITLE
Add full support for if let chain feature

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rinja-rs/rinja/issues/176.

Interestingly enough, the most difficult part was actually to ensure that variables creation was done in the right order.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Just thought that the "auto bool" cast might be missing. Gonna check later on.

**Comment by GuillaumeGomez** [comment]
And done.

File: rinja_parser/src/node.rs [hunk 1]
```diff
@@ -420,7 +420,19 @@ impl<'a> CondTest<'a> {
                 ws(|i: &mut _| Target::parse(i, s)),
                 ws('='),
             )),
-            ws(|i: &mut _| Expr::parse(i, s.level.get(), false)),
+            ws(|i: &mut _| {
+                let start = *i;
+                let mut expr = Expr::parse(i, s.level.get(), false)?;
+                if let Expr::BinOp(_, _, ref mut right) = expr.inner {
+                    if matches!(right.inner, Expr::Var("set" | "let")) {
+                        *i = &start[start.len() - i.len() - 3..];
```
Comment (Kijewski) [inline rinja_parser/src/node.rs:428]:
```suggestion
                        *i = right.span.as_suffix_of(start).unwrap();
```

**Comment by Kijewski** [comment]
Except for my one comment, the implementation looks good! A much smaller change than I would have anticipated.

File: rinja_parser/src/node.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply rinja_parser/src/node.rs]: Oh that's fancy!

**Comment by GuillaumeGomez** [comment]
Applied suggestion. :)

**Review by Kijewski — APPROVED** [review]
:+1:

**Pull request merged**